src/coreComponents/linearAlgebra/DofManager.hpp

```diff
@@ -71,6 +71,25 @@ class DofManager
     string meshLevelName;
     /// list of the region names
     std::vector< string > regionNames;
+
+    bool add( Regions const & input )
+    {
+      bool added = false;
+      if( meshBodyName  == input.meshBodyName && meshLevelName == input.meshLevelName )
+      {
+        for( auto const & inputRegionName : input.regionNames )
+        {
+          if( std::find( regionNames.begin(), regionNames.end(), inputRegionName ) == regionNames.end() )
+          {
+            regionNames.push_back( inputRegionName );
+            std::sort( regionNames.begin(), regionNames.end() );
+          }
```

Comment:
maybe `regionNames` should be a set?

Reply:
yeah...I was thinking that too. It's just requires more changes to some functions in `DofManager`. If we do that it would be nice to also find a better name to this struct `Regions` since it now contains also `meshBody` and `meshLevel`... Maybe it should be called `FieldSupport`?

Reply:
Furthermore, we have `FieldIdentifiers`, `MeshObjectPath` and `DofManager::Regions`, which while slightly different, all look alike. Could we rethink those?

Reply:
I feel the `MeshObjectPath` has kind of its own functionality linked to how we also specify b.c. etc. `FieldIdentifiers` and `DofManager::FieldSupport` (I am changing the name) are indeed very similar and could, probably be consolidated into a single thing.

Reply:
Although maybe `FieldDescription` is even similar.

Reply:
Indeed. We may try to at least unite them in some kind of file/namespace?

Reply:
Let me close this PR with the functionality I need and then I will open a small to try out merging 1, 2 or even all these things into a single object